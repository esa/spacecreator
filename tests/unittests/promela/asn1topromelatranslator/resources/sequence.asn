MYMODULE DEFINITIONS ::= BEGIN

EnvIntParam ::= INTEGER(0 .. 9)
EnvBoolParam ::= BOOLEAN

EnvParamSeq ::= SEQUENCE {
   x EnvIntParam,
   y EnvIntParam OPTIONAL,
   z EnvBoolParam
}

END