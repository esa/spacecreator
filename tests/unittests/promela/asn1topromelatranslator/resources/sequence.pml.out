inline EnvIntParam_generate_value(EnvIntParam_gv)
{
    if
    ::  atomic {
        select (EnvIntParam_gv : 0 .. 9);
    }
    fi;
}
inline EnvBoolParam_generate_value(EnvBoolParam_gv)
{
    if
    ::  true;
        EnvBoolParam_gv = 1;
    ::  true;
        EnvBoolParam_gv = 0;
    fi;
}
inline EnvParamSeq_generate_value(EnvParamSeq_gv)
{
    EnvIntParam_generate_value(EnvParamSeq_gv.x);
    if
    ::  true;
        EnvIntParam_generate_value(EnvParamSeq_gv.y);
        EnvParamSeq_gv.exist.y = 1;
    ::  true;
        EnvParamSeq_gv.exist.y = 0;
    fi;
    EnvBoolParam_generate_value(EnvParamSeq_gv.z);
}
